opensearch_py_ml/ml_models/sentencetransformermodel.py

```diff
@@ -1006,6 +1008,58 @@ def set_up_accelerate_config(
                 "Failed to open config file for ml common upload: " + file_path + "\n"
             )
 
+    def _get_model_description_from_md_file(self, readme_file_path) -> str:
+        """
+        Get description of the model from README.md file
+        after the model is saved in local directory
+
+        This function assumes that the md file has the following format in the README.md:
+
+        # sentence-transformers/msmarco-distilbert-base-tas-b
+        This is a port of the DistilBert TAS-B Model to sentence-transformers model:
+        It maps sentences & paragraphs to a 768 dimensional dense vector space and
+        is optimized for the task of semantic search.
+
+        # Usage (Sentence-Transformers)
+        ...
+
+        where "# MODEL_ID" is the first line that starts with #.
+
+        :param readme_file_path: Path to README.md file
+        :type readme_file_path: string
+        :return: Description of the model
+        :rtype: string
+        """
+        readme_data = MarkDownFile.read_file(readme_file_path)
+
+        # Find the description section
+        start_str = f"# {self.model_id}"
+        start = readme_data.find(start_str)
+        if start == -1:
+            model_name = self.model_id.split("/")[1]
+            start_str = f"# {model_name}"
+            start = readme_data.find(start_str)
+        end = readme_data.find("## ", start)
```

Comment:
Did you try with multiple sentence transformer models, how does this work. I feel like this is brittle. But I understand for the scraping we might need to depend on some placeholder. I just wanted to make sure if you checked with multiple different models?

Reply:
I randomly tried with many different models and it works. I can also add that it has to start with the word "This is". All the models I have seen use this format.

Reply:
Talked to Dhrubo offline and ended up not checking "This is" and made edits in the section ending part instead.